client/x/evmengine/types/tx.go

```diff
@@ -47,3 +51,73 @@ func (l *EVMEvent) Verify() error {
 
 	return nil
 }
+
+// EthLogToEVMEvent converts an Ethereum Log to an EVMEvent.
+func EthLogToEVMEvent(ethLog ethtypes.Log) (*EVMEvent, error) {
+	topics := make([][]byte, 0, len(ethLog.Topics))
+	for _, t := range ethLog.Topics {
+		topics = append(topics, t.Bytes())
+	}
+
+	evmEvent := &EVMEvent{
+		Address: ethLog.Address.Bytes(),
+		Topics:  topics,
+		Data:    ethLog.Data,
+	}
+	if err := evmEvent.Verify(); err != nil {
+		return nil, errors.Wrap(err, "verify log")
+	}
+
+	return evmEvent, nil
+}
+
+// SortEVMEvents sorts EVM events by Address > Topics > Data.
+func SortEVMEvents(events []*EVMEvent) {
+	sort.Slice(events, func(i, j int) bool {
+		if cmp := bytes.Compare(events[i].Address, events[j].Address); cmp != 0 {
+			return cmp < 0
+		}
+
+		topicI := slices.Concat(events[i].Topics...)
+		topicJ := slices.Concat(events[j].Topics...)
+		if cmp := bytes.Compare(topicI, topicJ); cmp != 0 {
+			return cmp < 0
+		}
+
+		return bytes.Compare(events[i].Data, events[j].Data) < 0
+	})
+}
+
+// IsSortedEVMEvents check if the events are sorted by ascending order of address, topics, and data.
+func IsSortedEVMEvents(events []*EVMEvent) bool {
```

Comment:
Is this only used in the test? If so, we should move it to the test

Reply:
@edisonz0718 

applied it in https://github.com/piplabs/story/pull/106/commits/828cb405aa5786f3324a346a24d55b965cb1cf03

thanks for your comment.

Reply:
@edisonz0718 
Could you review this PR when you have time? I applied your comment. Thanks in advance.